File: 1537-maximum-score-after-splitting-a-string/1537-maximum-score-after-splitting-a-string.cs
public class Solution {
    public int MaxScore(string s) {
        int ones = 0;
        foreach (char ch in s)
            if (ch == '1')
                ones++;
            
        
        int ans = 0;
        int zeros = 0;
        for (int i = 0; i < s.Length - 1; i++) {
            if (s[i] == '1') {
                ones--;
            } else {
                zeros++;
            }
            
            ans = Math.Max(ans, zeros + ones);
        }
        
        return ans;
    }
}

File: 1537-maximum-score-after-splitting-a-string/1537-maximum-score-after-splitting-a-string.go
func Max (l int, r int) int{
    if l >= r {
        return l
    }
    return r
}
func maxScore(s string) int {
    o := 0
    for _, ch := range s {
        if ch == '1' {
            o++
        }
    }

    n := len(s)
    ans, z := 0, 0
    for i := 0; i < n - 1; i++ {
        if s[i] == '1' {
            o--
        } else {
            z++
        }
        ans = Max(ans, z + o)
    }
    return ans
}